src/org/ods/orchestration/BuildStage.groovy

```diff
@@ -76,6 +79,12 @@ class BuildStage extends Stage {
         }
         executeInParallel(executeRepos, generateDocuments)
         levaDocScheduler.run(phase, MROPipelineUtil.PipelinePhaseLifecycleStage.PRE_END)
+
+        // in case of WIP we fail AFTER all pieces have been executed - so we can report as many
+        // failed unit tests as possible
+        if (project.isAssembleMode && project.isWorkInProgress && project.hasFailingTests) {
```

Comment:
@metmajer extend with a Check against all repos .. if there is a failed stage..

Reply:
@metmajer  - I added what we talked about - please verify .. I don't have a good testcase at hand